_posts/2020-01-12-updating-foreman-inventory-with-system-facts.md

```diff
@@ -0,0 +1,311 @@
+
+To customize the fact importing behavior, the following settings can be tweaked. Note that they
+apply to every fact source, despite the fact, some of them are listed under
+**Puppet** and **Provisioning** settings category.
+
+<table class="table table-bordered table-condensed">
+  <thead>
+    <tr>
+      <th>Setting</th>
+      <th>Category</th>
+      <th>Effect</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>Create new host when facts are uploaded</td>
+      <td>Puppet</td>
+      <td>If set to true (default), if facts for an unknown hosts are received, Host object will be created based on them</td>
+    </tr>
+    <tr>
+      <td>Create new host when report is uploaded</td>
+      <td>Puppet</td>
+      <td>If set to true (default), if a report for an unknown host is received, Host object will be created based on it. Report typically contains less information. Report usually contains information about what has happened during the configuration deployment, but at least the host name is available</td>
+    </tr>
+    <tr>
+      <td>Default location</td>
+      <td>Puppet</td>
+      <td>When host is created based on either facts, we search for a fact representing the location. If there's no such fact or the host  is created based on the report, it will be assigned to the location specified in this setting.</td>
+    </tr>
+    <tr>
+      <td>Default organization</td>
+      <td>Puppet</td>
+      <td>Same as default location, but works for organization</td>
+    </tr>
+    <tr>
+      <td>Location fact</td>
+      <td>Puppet</td>
+      <td>The name of the fact, which contains the name of the location, in which this host should be placed. By default, there's no such  fact, but all configuration management systems allow to specify custom fact. Hence you can create a custom fact e.g. foreman_location and then the Host will be automatically assigned to it. </td>
+    </tr>
+    <tr>
+      <td>Organization fact</td>
+      <td>Puppet</td>
+      <td>Same as above but for organization</td>
+    </tr>
+    <tr>
+      <td>Update subnets from facts</td>
+      <td>Puppet</td>
+      <td>If an IP of existing interface is changed (or newly created), Foreman will try to match existing Subnet object based on this setting.</td>
+    </tr>
+    <tr>
+      <td>Exclude pattern for facts stored in foreman</td>
+      <td>Provisioning</td>
+      <td>Facts can contain a lot of data that may not be interesting. E.g. if you don't want to import all blockdevices from each host, all network interfaces etc, you can specify a pattern for ignoring. The default value contains commonly ignored patterns, e.g. "docker&ast;"</td>
+    </tr>
+    <tr>
+      <td>Ignore facts for domain</td>
+      <td>Provisioning</td>
+      <td>By default, Foreman will be creating and assigning Domain objects based on the host domain name. If you don't want this to be updated, you can disable this behavior using this setting.</td>
+    </tr>
```

Comment:
Domain objects reads weird, it's Host Domain perhaps.

Reply:
yeah, though we use it to distinguish between host domain and the Domain resource (active record model) we have in Foreman under Infrastructure -> Domains